src/c++/perf_analyzer/report_writer.cc

```diff
@@ -388,4 +407,37 @@ ReportWriter::WriteGpuMetrics(std::ostream& ofs, const Metrics& metric)
   }
 }
 
+void
+ReportWriter::WriteLlmMetrics(std::ostream& ofs)
```

Comment:
this function seems to calculate AND write. Sounds like 2 functions to me.

Reply:
Extracted calculation part out to separate function